### PR TITLE
Add filter box that appears above take a stroll button

### PR DESCRIPTION
**Pull request by carpenterd777**
In addition to doing as the title says, this also adds a few more components to Peer's component library - Box and CheckBox.

### CODE REVIEW

**Review by elenirotsides — CHANGES_REQUESTED** [review]
Blocking this for now until I can run this on iOS. The app crashes when I click on the filter button. The error I get in my terminal:

```
Invariant Violation: requireNativeComponent: "RNCCheckbox" was not found in the UIManager.

This error is located at:
    in RNCCheckbox (created by CheckBox)
    in RCTView (at View.js:34)
    in View (created by CheckBox)
    in CheckBox (created by ForwardRef(CheckBoxWithRef))
    in ForwardRef(CheckBoxWithRef) (at CheckBox.tsx:28)
    in RCTView (at View.js:34)
    in View (at CheckBox.tsx:24)
    in CheckBox (at FilterBox.tsx:31)
    in RCTView (at View.js:34)
    in View (at Box.tsx:15)
    in Box (at FilterBox.tsx:25)
    in FilterBox (at MainView.tsx:14)
    in RCTView (at View.js:34)
    in View (at MainView.tsx:13)
    in RCTView (at View.js:34)
    in View (at MainView.tsx:11)
    in MainView (at App.tsx:5)
    in App (created by ExpoRoot)
    in ExpoRoot (at renderApplication.js:45)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:106)
    in DevAppContainer (at AppContainer.js:121)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:132)
    in AppContainer (at renderApplication.js:39)
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:104:6 in reportException
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:171:19 in handleException
at node_modules/react-native/Libraries/Core/ReactFiberErrorDialog.js:43:2 in showErrorDialog

Invariant Violation: requireNativeComponent: "RNCCheckbox" was not found in the UIManager.

This error is located at:
    in RNCCheckbox (created by CheckBox)
    in RCTView (at View.js:34)
    in View (created by CheckBox)
    in CheckBox (created by ForwardRef(CheckBoxWithRef))
    in ForwardRef(CheckBoxWithRef) (at CheckBox.tsx:28)
    in RCTView (at View.js:34)
    in View (at CheckBox.tsx:24)
    in CheckBox (at FilterBox.tsx:31)
    in RCTView (at View.js:34)
    in View (at Box.tsx:15)
    in Box (at FilterBox.tsx:25)
    in FilterBox (at MainView.tsx:14)
    in RCTView (at View.js:34)
    in View (at MainView.tsx:13)
    in RCTView (at View.js:34)
    in View (at MainView.tsx:11)
    in MainView (at App.tsx:5)
    in App (created by ExpoRoot)
    in ExpoRoot (at renderApplication.js:45)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:106)
    in DevAppContainer (at AppContainer.js:121)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:132)
    in AppContainer (at renderApplication.js:39)
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:104:6 in reportException
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:171:19 in handleException
at node_modules/react-native/Libraries/Core/ReactFiberErrorDialog.js:43:2 in showErrorDialog

Invariant Violation: requireNativeComponent: "RNCCheckbox" was not found in the UIManager.

This error is located at:
    in RNCCheckbox (created by CheckBox)
    in RCTView (at View.js:34)
    in View (created by CheckBox)
    in CheckBox (created by ForwardRef(CheckBoxWithRef))
    in ForwardRef(CheckBoxWithRef) (at CheckBox.tsx:28)
    in RCTView (at View.js:34)
    in View (at CheckBox.tsx:24)
    in CheckBox (at FilterBox.tsx:31)
    in RCTView (at View.js:34)
    in View (at Box.tsx:15)
    in Box (at FilterBox.tsx:25)
    in FilterBox (at MainView.tsx:14)
    in RCTView (at View.js:34)
    in View (at MainView.tsx:13)
    in RCTView (at View.js:34)
    in View (at MainView.tsx:11)
    in MainView (at App.tsx:5)
    in App (created by ExpoRoot)
    in ExpoRoot (at renderApplication.js:45)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:106)
    in DevAppContainer (at AppContainer.js:121)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:132)
    in AppContainer (at renderApplication.js:39)
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:104:6 in reportException
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:171:19 in handleException
at node_modules/react-native/Libraries/Core/ReactFiberErrorDialog.js:43:2 in showErrorDialog

Invariant Violation: requireNativeComponent: "RNCCheckbox" was not found in the UIManager.

This error is located at:
    in RNCCheckbox (created by CheckBox)
    in RCTView (at View.js:34)
    in View (created by CheckBox)
    in CheckBox (created by ForwardRef(CheckBoxWithRef))
    in ForwardRef(CheckBoxWithRef) (at CheckBox.tsx:28)
    in RCTView (at View.js:34)
    in View (at CheckBox.tsx:24)
    in CheckBox (at FilterBox.tsx:31)
    in RCTView (at View.js:34)
    in View (at Box.tsx:15)
    in Box (at FilterBox.tsx:25)
    in FilterBox (at MainView.tsx:14)
    in RCTView (at View.js:34)
    in View (at MainView.tsx:13)
    in RCTView (at View.js:34)
    in View (at MainView.tsx:11)
    in MainView (at App.tsx:5)
    in App (created by ExpoRoot)
    in ExpoRoot (at renderApplication.js:45)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:106)
    in DevAppContainer (at AppContainer.js:121)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:132)
    in AppContainer (at renderApplication.js:39)
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:104:6 in reportException
at node_modules/react-native/Libraries/Core/ExceptionsManager.js:171:19 in handleException
at node_modules/react-native/Libraries/Core/setUpErrorHandling.js:24:6 in handleError
at node_modules/expo-error-recovery/build/ErrorRecovery.fx.js:12:21 in ErrorUtils.setGlobalHandler$argument_0
```

File: src/components/CheckBox.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	style: StyleProp<object>;
+}
+
+const CheckBox = ({
+	value,
+	onValueChange,
+	text,
+	accessibilityLabel,
+	style,
+}: CheckBoxProps): JSX.Element => {
```
Comment (aej11a) [inline src/components/CheckBox.tsx:22]: Same comment about React.FC

File: src/components/CheckBox.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { View, Text, StyleSheet } from "react-native";
+import type { StyleProp } from "react-native";
+import * as CB from "@react-native-community/checkbox";
```
Comment (aej11a) [inline src/components/CheckBox.tsx:4]: `import CB from "`

File: src/components/Button.tsx [hunk 3]
```diff
@@ -10,7 +11,7 @@ export interface ButtonProps {
 	accessibilityLabel: string; // not optional for this project.
 	onPress: VoidFunction;
 	//eslint-disable-next-line @typescript-eslint/ban-types
-	style?: object; // may be unsafe, but this is the type provided
+	style?: StyleProp<object>; // may be unsafe, but this is the type provided
```
[the remark this answers is not in the document]
Comment (carpenterd777) [reply src/components/Button.tsx:14]: eslint still doesn't like the usage of the `object` type. I'm not sure what the "actual" type that goes in the generic here is though.

File: src/components/StrollButton.tsx [hunk 4]
```diff
@@ -1,16 +1,30 @@
...
 import { Button } from "./Button";
 
-const StrollButton: React.FC<{ onStartStrolling: () => void }> = ({ onStartStrolling }) => {
-	const [isShowingFilters, setIsShowingFilters] = useState<boolean>(false);
+export interface StrollButtonProps {
+	onStartStrolling: () => void;
+	isShowingFilters: boolean;
+	setIsShowingFilters: React.Dispatch<React.SetStateAction<boolean>>;
+	//eslint-disable-next-line @typescript-eslint/ban-types
+	style?: StyleProp<object>; // may be unsafe, but this is the type provided
```
[the remark this answers is not in the document]
Comment (carpenterd777) [reply src/components/StrollButton.tsx:13]: mmm its actually not quite true anymore. the documentation lists it as being an `object`, but it also like it better when we use `StyleProp` in some way. I'll update all instances of this comment to reflect that information

File: src/components/Button.tsx [hunk 5]
```diff
@@ -1,26 +1,25 @@
...
+	style?: StyleProp<object>; // TODO: update generic from "object"
 }
 
 /**
  * Peer-styled button element.
  */
 export const Button: React.FC<ButtonProps> = ({
 	text,
-	image,
+	iconName: image,
```
Comment (aej11a) [inline src/components/Button.tsx:22]: This can be fixed later but at this point i would just get rid of "image" entirely

**Pull request merged**